controllers/glanceapi_controller.go

```diff
@@ -178,6 +184,53 @@ func (r *GlanceAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *GlanceAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// index passwordSecretField
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &glancev1.GlanceAPI{}, passwordSecretField, func(rawObj client.Object) []string {
```

Comment:
This part could be partially optimized by iterating over `glanceAPIWatchFields` and index the current field, but I'm not sure how the resulting  `cr.Spec.<field>` can be dynamically resolved.